hack/e2e/ensure_go.sh

```diff
@@ -2,7 +2,7 @@
 
 set -eux
 
-MINIMUM_GO_VERSION=go1.20.14
+MINIMUM_GO_VERSION=go1.21.7
```

Comment:
Not sure if we should bump this? We've been bumping Dockerfile + Makefile only for others?

Reply:
removed it